io-engine-tests/src/lib.rs

```diff
@@ -541,3 +542,92 @@ macro_rules! test_diag {
         println!($($arg)*);
     }}
 }
+
+pub fn create_zoned_nullblk_device(
+    block_size: u32,
+    zone_size: u32,
+    zone_cap: u32,
+    nr_conv_zones: u32,
+    nr_seq_zones: u32,
+    max_active_zones: u32,
+    max_open_zones: u32,
+) -> Result<u32, (i32, String)> {
```

Comment:
Could we return a new type Wrapper with RAII instead? example:

```suggestion
) -> Result<NullBlk, (i32, String)> {
```

```rust
struct NullBlk(u32);
impl Drop for NullBlk {
  fn drop(&mut self) {
    delete_nullblk_device(self.0);
  }
}
```